setup.py

```diff
@@ -18,7 +18,7 @@
 
 # Dependencies
 # Server minded
-elastic_deps = ["elasticsearch-dsl~=7.4,<8.0"]
+elastic_deps = ["elasticsearch-dsl~=7.4,<8.0", "elasticsearch~=7.17"]
```

Comment:
I think so, yes. Different versions of `elasticsearch` are introducing different functionality that `elasticsearch-dsl` is making use of in different ways. Currently DLS is making deprecated calls to the underlying package without pinning the version, so if we want to avoid deprecation warnings ourselves then I think this is the only choice.

Reply:
Do you mean that we get deprecation warnings because the elasticsearch version is too old ?
And that elasticsearch it is not automatically updated when we update elasticsearch-dsl ?

Reply:
We get them because `elasticsearch_dsl` is using the Python API defined in `elasticsearch` (the Python package) but is not completely up to date itself (and does not pin a specific version as it is a library).

Reply:
So does Elasticsearch-dsl give you a deprecation warning when you use an older version(e.g. 7.1) of Elasticsearch ?
Anyway, I do not think this is a big enough issue to hold back this PR, so I have approved it.

Reply:
The Python client `elasticsearch` (not the database) follows the same version convention as the database, but `elasticsearch_dsl` (the higher-level interface) just uses the major version to say they support v7 of the database. We were directly making calls to the low-level `elasticsearch` Python API (without adding it as an explicit dependency, until this PR) that have been deprecated (fixed in this PR), but the DSL is still making similar deprecated calls that we cannot fix at our end - I have added these to the pytest ignore list. As `elasticsearch-dsl` does not tightly pin the version of `elasticsearch` Python API, there is a chance that those calls get removed in a later version, which would also break our package unless we pin it.